libs/sdk-ui-kit/src/Header/generateHeaderMenuItemsGroups.ts

```diff
@@ -0,0 +1,142 @@
+// (C) 2007-2021 GoodData Corporation
+import { IFeatureFlags } from "@gooddata/goodstrap/lib/@next/Header/types";
+import { IHeaderMenuItem } from "@gooddata/goodstrap/lib/@next";
+import { IWorkspacePermissions } from "../typings/featureFlags";
+import { isFreemiumEdition, shouldEnableNewNavigation, shouldHidePPExperience } from "../utils/featureFlags";
+
+/**
+ * @internal
+ */
+export function generateHeaderMenuItemsGroups(
+    targetApp: "kpiDashboards" | "analyticalDesigner",
```

Comment:
if this function is parametrized in this way, I would rather move it into each app's repo. How is it extensible for Load and other new applications?

Reply:
What about moving logic using targetApp param to separate function implemented in each app and keep of generic generator here to don't repeat the common code?

Reply:
Yes, agreed, this has no place in the SDK, code specific to an app should be in that app.

Reply:
After discussion with team, I will remove this parameter and all related logic code will be moved into app instead of library. I just keep a common logic for all app. Thanks